src/axom/inlet/examples/nested_structs.cpp

```diff
@@ -0,0 +1,268 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/inlet.hpp"
+
+#include <iostream>
+
+#include "fmt/fmt.hpp"
+
+#include "axom/slic/core/SimpleLogger.hpp"
+
+namespace inlet = axom::inlet;
+using Vector = inlet::FunctionType::Vec3D;
+
+// Used to convert an unordered map (the type Inlet uses) to represent arrays
+// to a geometric vector type
+// FIXME: Clean this up/remove when PR #429 is merged
+Vector mapToVector(const std::unordered_map<int, double>& map)
+{
+  Vector result;
+  // Make sure the elements are accessed in order
+  for(int i = 0; i < 3; i++)
+  {
+    const auto ele = map.find(i);
+    if(ele != map.end())
+    {
+      result[i] = ele->second;
+    }
```

Comment:
Should it be an error if the element is missing? I'm not sure what the expected usage of this is, so I can't tell for sure. Noticing that it's in an example, though, it might not really matter.

Reply:
In theory, yes, but to avoid cluttering the example with logic that will be added in #429 and #426, I have omitted some verification logic here.